middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { withAuth } from "next-auth/middleware";
+// middleware.ts
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "next-auth/jwt";
+
+export async function middleware(req: NextRequest) {
+  // Skip middleware for static files and API routes
+
+  const session = await getToken({ req, secret: process.env.NEXTAUTH_SECRET });
+  const { pathname } = req.nextUrl;
+
+  // Redirect to sign-up if there's no session and not on the sign-up page
+  if (!session && pathname !== "/sign-up") {
+    const url = req.nextUrl.clone();
+    url.pathname = "/sign-up";
+    return NextResponse.redirect(url);
+  }
```

Comment:
```suggestion
  // Redirect to sign-up if there's no session
  if (!session) {
    const url = req.nextUrl.clone();
    url.pathname = "/sign-up";
    return NextResponse.redirect(url);
  }
```


If we have simply don't protect the sign-in and sign-up routes, we can avoid checking to make sure we're not on them first.

Also with these changes we can actually access the sign-in page :)

Reply:
10-4!